### PR TITLE
Save Organization.available_features as a DB column

### LINKED ISSUE #4408: Make `Organization.available_features` available to the plugin server
## Is your feature request related to a problem?

Currently detecting features available for an organization is a complex affair involving queries on 3 models (1 is FOSS, 1 is EE, and 1 is Cloud). This makes getting these features in the plugin server infeasible, and we need them to run Zapier there (per https://github.com/PostHog/plugin-server/issues/235), which is a premium feature.

## Describe the solution you'd like

TBD.

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

Closes #4408 (a prerequisite for https://github.com/PostHog/plugin-server/issues/235).

Makes `Organization.available_features` a model field (Postgres column) instead of a dynamically calculated property. This way the features-determination logic only has to run occasionally, and the results are available in the database, so the plugin server can use them too.

The field is updated (with `organization.update_available_features()`) in the following circumstances:
- when the server is started, all orgs at once – in case the predefined plans just changed
- every hour half past the hour, all orgs at once – in case of subscription changes/license expiration, eventual consistency sort of
- when a License instance is created/updated, all orgs at once – for private instances
- when an OrganizationBilling instance is created/updated, only the relevant org – for Cloud (https://github.com/PostHog/posthog-cloud/pull/121)

## Checklist

- [x] All querysets/queries filter by Organization, by Team, and by User
- [x] Django backend tests
- [x] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious

### CODE REVIEW

File: posthog/tasks/sync_all_organization_available_features.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from typing import Sequence, cast
+
+from posthog.models.organization import Organization
+
+
+def sync_all_organization_available_features() -> None:
```
Comment (paolodamico) [inline posthog/tasks/sync_all_organization_available_features.py:6]: Nit: naming `sync_all_organizations_available_features`

File: posthog/models/organization.py [hunk 2]
```diff
@@ -80,6 +81,7 @@ class PluginsAccessLevel(models.IntegerChoices):
         default=PluginsAccessLevel.CONFIG if settings.MULTI_TENANCY else PluginsAccessLevel.ROOT,
         choices=PluginsAccessLevel.choices,
     )
+    available_features = ArrayField(models.CharField(max_length=64, null=False, blank=False), blank=True, default=list)
```
Comment (paolodamico) [inline posthog/models/organization.py:84]: Nit: `null=False` is already the default behavior.

File: posthog/models/organization.py [hunk 3]
```diff
@@ -148,8 +152,14 @@ def get_analytics_metadata(self):
         }
 
 
+@receiver(models.signals.pre_save, sender=Organization)
+def organization_about_to_be_created(sender, instance: Organization, raw, using, **kwargs):
+    if instance._state.adding:
+        instance.update_available_features()
```
Comment (paolodamico) [inline posthog/models/organization.py:158]: Maybe worth just updating this particular org?

**Review by paolodamico — APPROVED** [review]
lgtm! minor comments

File: posthog/models/organization.py (hunk 3 above)
Comment (Twixes) [reply posthog/models/organization.py:158]: WDYM? That does seem to be the case already?

File: posthog/tasks/sync_all_organization_available_features.py (hunk 1 above)
Comment (Twixes) [reply posthog/tasks/sync_all_organization_available_features.py:6]: This is more of "Sync all `Organization.available_features`", I think I'll leave as is

**Pull request merged**